qtconsole/client.py

```diff
@@ -45,6 +45,10 @@ def call_handlers(self, msg):
         # Emit the generic signal.
         self.message_received.emit(msg)
 
+    def closed(self):
+        """Check if the channel is closed."""
+        return self.stream.closed()
```

Comment:
After it finishes closing, the channel sets it to None, so handle that, too:

```suggestion
        return self.stream is None or self.stream.closed()
```